Task NXDOC-1934 add documentation (#18)
nuxeo/doc.nuxeo.com-client-java: src/client-java/authentication.md

```
There's currently no way to obtain a JWT token through built-in REST API.

> ## OAuth2 Authentication

Reviewer: ```suggestion
## OAuth 2 Authentication
```

### With An Authorization Flow

> OAuth2 authentication is enabled by default on Nuxeo server, you just need to create an OAuth2 Client in Nuxeo, see [Nuxeo - OAuth2 Client Registration]({{page space='nxdoc' page='using-oauth2'}}#client-registration).

Reviewer: ```suggestion
OAuth 2 authentication is enabled by default on Nuxeo server, you just need to create an OAuth 2 Client in Nuxeo, see [Nuxeo - OAuth2 Client Registration]({{page space='nxdoc' page='using-oauth2'}}#client-registration).
```

> Once you have your `clientId` and `clientSecret` you need to implement an [OAuth2 Grant Flow]({{page space='nxdoc' page='using-oauth2'}}#oauth-2-authorization-grant-flow) in your application to retrieve an authentication code. With this code you can setup the OAuth2 authentication in the java client with:

Reviewer: ```suggestion
Once you have your `clientId` and `clientSecret` you need to implement an [OAuth 2 Grant Flow]({{page space='nxdoc' page='using-oauth2'}}#oauth-2-authorization-grant-flow) in your application to retrieve an authentication code. With this code you can setup the OAuth 2 authentication in the Java client with:
```

```java
new NuxeoClient.Builder().authentication(
    OAuth2AuthInterceptor.obtainAuthFromAuthorizationCode("http://localhost:8080/nuxeo", clientId, authCode));
// or if you have set up a client with a clientSecret
new NuxeoClient.Builder().authentication(
    OAuth2AuthInterceptor.obtainAuthFromAuthorizationCode("http://localhost:8080/nuxeo", clientId, clientSecret, authCode));
```

> After that, your client will be able to access Nuxeo using OAuth2 and it will refresh the OAuth2 token if needed.

Reviewer: ```suggestion
After that, your client will be able to access Nuxeo using OAuth 2 and it will refresh the OAuth 2 token if needed.
```

### With an Access Token

> You can also use OAuth2 authentication within java client with an access token. This is useful when the software using the client is responsible to obtain and refresh the OAuth Token.

Reviewer: ```suggestion
You can also use OAuth 2 authentication within the Java client with an access token. This is useful when the software using the client is responsible to obtain and refresh the OAuth Token.
```
